application/helpers/common_helper.php

```diff
@@ -5470,3 +5473,75 @@ function csvEscape($string)
 
     return $string;
 }
+
+/**
+ * Convert legacy INSERTANS tags to modern question code references
+ *
+ * Converts {INSERTANS:SIDXGIDXQID} format to questionTitle.shown format
+ * Examples:
+ * - {INSERTANS:554233X11X1} → G01Q02.shown (if QID 1 exists)
+ * - {INSERTANS:554233X11X11} → .shown (if QID 11 doesn't exist)
+ * - {INSERTANS:554233X11X1someText} → someText.shown (if QID 1 exists)
+ * - {INSERTANS:554233X11X11someText} → .shown (if QID 11 doesn't exist)
+ * - {INSERTANS:554233X11Xother} → other.shown (letters only)
+ *
+ * @param string $text The text containing INSERTANS tags
+ * @param int $oldSid The old survey ID to look up questions from
+ * @return string The text with INSERTANS tags converted
+ */
+function convertLegacyInsertans($text, $oldSid)
+{
+    $txtInsertans = "{INSERTANS:";
+
+    if (strpos($text, $txtInsertans) === false) {
+        return $text;
+    }
+
+    // 1. Extract QIDs after last X
+    preg_match_all('/\{INSERTANS:([^}]*)\}/', $text, $blocks);
```

Comment:
This is costly for long strings, check for the first position of $txtInsertans first and return $text if the result is false, which will be the normal case. And please reuse $txtInsertans inside your preg_match_all call.